app/data_model/answer_store.py

```diff
@@ -59,16 +61,32 @@ class AnswerStore:
 
     def __init__(self, existing_answers=None):
         self.answers = existing_answers or []
+        self.answer_map = self._build_map(self.answers)
```

Comment:
I think we should consider removing the `self.answers` list completly and just ising the map.
We should also look at the gains of storing the map in the datastore rather than converting a map to and from a list